Q12.py

```diff
@@ -0,0 +1,45 @@
+class BankAccount:
+    def __init__(self, acc_name, acc_number, balance, pin):
+        self.acc_name = acc_name
+        self.acc_number=acc_number
+        self.balance=balance
+        self.pin=pin
+    
+    def check_pin(self, pin):
+        #Checking the pin on bases of condition and then giving account information
+        if self.pin == pin:
+            return "Account Holder Name: {}\nAccount Number: {}\nTotal Balance: {}".format(self.acc_name, self.acc_number, self.balance)
+        return "Please Enter Correct Pin"
+    
+    def deposit(self, pin, amount):
+
+        #Checking the pin on bases of condition and then depositing money in  account.
+        if (self.pin == pin):
+            self.balance += amount
```

Comment:
Why you are repeating this lines multiple times? You again didn't checked your code properly. You've already created check_pin function but same logic you are writing here. Please update it.